lib/Widgets/SeekBar.vala

```diff
@@ -16,34 +16,232 @@
  */
 
 /**
- * The SeekBar is a widget for viewing and setting media playback position.
+ * This widget is a playback statusbar that contains a gtk.scale widget and
+ * two labels showing the current time and the time remaining.
  *
  * Granite.SeekBar will get the style class .app-notification
  *
- * {{../../doc/images/Toast.png}}
+ * {{../../doc/images/SeekBar.png}}
  */
 
 public class Granite.Widgets.SeekBar : Gtk.Grid {
+    /*
+     * The time of the full duration of the playback.
+     */
+    public double playback_duration { get; construct set; }
+
+    /*
+     * The progression of the playback as a decimal from 0.0 to 1.0.
+     */
+    public double playback_progress { get; private set; }
+
+    /*
+     * If the pointer is grabbing the scale button.
+     */
+    public bool is_grabbing { get; private set; }
+
+    /*
+     * If the pointer is hovering over the scale.
+     */
+    public bool is_hovering { get; private set; }
+
+
+    /*
+     * The left label that displays the time progressed.
+     */
+    public Gtk.Label progression_label { get; private set; }
+
+    /*
+     * The right label that displays the total duration time.
+     */
+    public Gtk.Label time_label {  get; private set;  }
+
+    /*
+     * The time of the full duration of the playback.
+     */
+    public Gtk.Scale scale {  get; private set;  }
 
-    /**
-     * Creates a new SeekBar
+
+    /*
+     * Emitted when the scale's window is hovered over.
+     */
+    public signal void scale_hover (Gdk.EventCrossing event);
```

Comment:
I don't think there is a need for this `scale_` API. The application using SeekBar can just access `scale` widget and operate / connect to it's signals itself without any overhead.